src/StatusQ/Components/StatusExpandableSettingsItem.qml

```diff
@@ -0,0 +1,150 @@
+import QtQuick 2.14
+import StatusQ.Core 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+
+Rectangle {
+    id: root
+
+    property alias icon: icon
+    property alias primaryText: primaryText
+    property alias secondaryText: secondaryText
+    property alias button: button
+    property alias expandableComponent: expandableRegion.sourceComponent
+
+    property bool isExpandable: true
+    property bool hasButton: false
+    property  bool isExpanded: false
+    property bool hasDropDownArrow: false
+
+    anchors.left: parent.left
+    anchors.leftMargin: 20
+    anchors.right: parent.right
+    anchors.rightMargin: 20
+
+    radius: 8
+    color: Theme.palette.indirectColor1
+    border.color: Theme.palette.baseColor2
+    state: "COLLAPSED"
+    clip: true
+
+    StatusRoundIcon {
+        id: icon
+        anchors.top: parent.top
+        anchors.topMargin: 25
+        anchors.left: parent.left
+        anchors.leftMargin: 11
+
+        icon.background.width: 32
+        icon.background.height: 32
+        icon.background.color: Theme.palette.primaryColor2
+        icon.color: Theme.palette.directColor1
+    }
+
+    StatusBaseText {
+        id: primaryText
+        anchors.top: parent.top
+        anchors.topMargin: 17
+        anchors.left: icon.right
+        anchors.leftMargin: 10
+        anchors.right: parent.right
+        anchors.rightMargin: 10
+
+        font.weight: Font.Medium
+        font.pixelSize: 15
+        lineHeight: 22
+        lineHeightMode: Text.FixedHeight
+        elide: Text.ElideRight
+        color: Theme.palette.directColor1
+    }
+
+    StatusBaseText {
+        id: secondaryText
+        anchors.top: primaryText.bottom
+        anchors.topMargin: 4
+        anchors.left: primaryText.left
+        anchors.right: primaryText.right
+
+        font.pixelSize: 15
+        lineHeight: 22
+        lineHeightMode: Text.FixedHeight
+        elide: Text.ElideRight
+        color: Theme.palette.directColor3
+    }
+
+    StatusButton {
+        id: button
+        anchors.top: parent.top
+        anchors.topMargin: 19
+        anchors.right: parent.right
+        anchors.rightMargin: 17
+        visible: root.hasButton
+    }
+
+    StatusIcon {
+        id: expandImage
+        anchors.top: parent.top
+        anchors.topMargin: 36
+        anchors.right: parent.right
+        anchors.rightMargin: 23
+        visible: hasDropDownArrow && isExpandable
+        color: Theme.palette.directColor1
+    }
+
+    Loader {
+        id: expandableRegion
+        active:  root.isExpanded
+        anchors.top: secondaryText.bottom
+        anchors.topMargin: 16
+        anchors.left: parent.left
+        anchors.leftMargin: 48
+        width: parent.width - 64
+    }
+
+    MouseArea {
+        anchors.fill: parent
+        enabled: !hasButton
```

Comment:
It looks like this isn't working properly.

When there's only a button and the component is not expandable, most likely only the button should be clickable, not the entire component. However, we still see the cursor turning into a pointing hand cursor.